component/src/main/java/com/synopsys/integration/alert/component/diagnostic/utility/RabbitMQDiagnosticUtility.java

```diff
@@ -0,0 +1,39 @@
+package com.synopsys.integration.alert.component.diagnostic.utility;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.springframework.amqp.core.AmqpAdmin;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.api.event.AlertMessageListener;
+import com.synopsys.integration.alert.component.diagnostic.model.AlertQueueInformation;
+import com.synopsys.integration.alert.component.diagnostic.model.RabbitMQDiagnosticModel;
+
+@Component
+public class RabbitMQDiagnosticUtility {
+    private final AmqpAdmin amqpAdmin;
+    private final Set<String> alertMessageListenerNames;
+
+    @Autowired
+    public RabbitMQDiagnosticUtility(AmqpAdmin amqpAdmin, List<AlertMessageListener<?>> allAlertMessageListeners) {
+        this.amqpAdmin = amqpAdmin;
+        this.alertMessageListenerNames = allAlertMessageListeners
+            .stream()
+            .map(AlertMessageListener::getDestinationName)
+            .collect(Collectors.toSet());
+    }
+
+    public RabbitMQDiagnosticModel getRabbitMQDiagnostics() {
+        List<AlertQueueInformation> alertQueueInformation = alertMessageListenerNames
+            .stream()
+            .map(amqpAdmin::getQueueInfo)
+            .filter(Objects::nonNull)
```

Comment:
Have you seen an instance of null returned?  If so do you know when this happens?

Reply:
QueueInfo is nullable, I added the check here for a sonar issues related to the problem.